indexer/eventprocessor.go

```diff
@@ -7,78 +7,111 @@ import (
 	"gorm.io/gorm"
 
 	"github.com/photon-storage/go-common/log"
+	"github.com/photon-storage/go-photon/crypto/sha256"
 
 	"github.com/photon-storage/photon-explorer/chain"
 	"github.com/photon-storage/photon-explorer/database/orm"
 )
 
 // EventProcessor is the processor for synchronizing photon change events.
 type EventProcessor struct {
+	ctx             context.Context
 	refreshInterval uint64
 	db              *gorm.DB
 	node            *chain.NodeClient
 	currentSlot     uint64
 	currentHash     string
+	quit            chan struct{}
 }
 
 // NewEventProcessor returns the new instance of EventProcessor.
-func NewEventProcessor(refreshInterval uint64, nodeEndpoint string, db *gorm.DB) *EventProcessor {
+func NewEventProcessor(
+	ctx context.Context,
+	refreshInterval uint64,
+	nodeEndpoint string,
+	db *gorm.DB,
+) *EventProcessor {
 	currentSlot, currentHash, err := currentChainStatus(db)
```

Comment:
It is better to move this fetch to Run() and retry until it succeeds. curentSlot and currentHash can be local var.